File: ports/rust/src/gui/layout/bcsboxlayout.rs
use super::bcslayout::BcsLayout;

pub enum BcsDirection {
    Horizontal,
    Vertical,
}

// BcsBoxLayout organizes widgets into a horizontal or vertical array
pub struct BcsBoxLayout {
    layout: BcsLayout,
    pub direction: BcsDirection,
}

impl BcsBoxLayout {
    pub fn new(direction: BcsDirection) -> Self {
        BcsBoxLayout {
            layout: BcsLayout::new(),
            direction,
        }
    }

    // Invalidate triggers an algorithmic recalculation of widget bounds
    pub fn invalidate(&mut self) {
        // Recompute sizes
    }
}

File: ports/go/src/gui/layout/bcsboxlayout.go
package layout

// BcsBoxLayout organizes widgets into a horizontal or vertical array
type BcsBoxLayout struct {
	*BcsLayout
	Direction int // 0 = Horizontal, 1 = Vertical
}

func NewBcsBoxLayout(dir int) *BcsBoxLayout {
	return &BcsBoxLayout{
		BcsLayout: NewBcsLayout(),
		Direction: dir,
	}
}

// Invalidate triggers an algorithmic recalculation of widget bounds
func (b *BcsBoxLayout) Invalidate() {
	// Recompute sizes
}
